ts/features/newProfile/saga/profile.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * A saga that manages the Profile.
+ */
+import * as E from "fp-ts/lib/Either";
+import { pipe } from "fp-ts/lib/function";
+import * as O from "fp-ts/lib/Option";
+import { delay, call, put, select, takeLatest } from "typed-redux-saga/macro";
+import Config from "react-native-config";
+import { getType } from "typesafe-actions";
+import { Millisecond } from "@pagopa/ts-commons/lib/units";
+import { InitializedProfile } from "../../../../definitions/backend/InitializedProfile";
+import { BackendClient } from "../../../api/backend";
+import I18n from "../../../i18n";
+import { sessionExpired } from "../../../store/actions/authentication";
+import {
+  profileLoadFailure,
+  profileLoadSuccess
+} from "../../../store/actions/profile";
+import { ReduxSagaEffect, SagaCallReturnType } from "../../../types/utils";
+import { convertUnknownToError } from "../../../utils/errors";
+import { readablePrivacyReport } from "../../../utils/reporters";
+import { sessionTokenSelector } from "../../../store/reducers/authentication";
+import { authenticationSaga } from "../../../sagas/startup/authenticationSaga";
+import { refreshUserProfileDataRequest } from "../store/actions/profile";
+
+const BACKEND_PROFILE_RETRY_DELAY_INTERVAL = (6 * 1000) as Millisecond;
+const apiUrlPrefix: string = Config.API_URL_PREFIX;
+
+// This function listens for Profile refresh requests and calls the needed saga.
+export function* refreshProfileOnMountOfTheNewProfilePageSaga(): Iterator<ReduxSagaEffect> {
+  yield* takeLatest(getType(refreshUserProfileDataRequest), refreshProfile);
+}
+
+export function* refreshProfile(): Generator<ReduxSagaEffect, void, any> {
+  // Whether the user is currently logged in.
+  const previousSessionToken: ReturnType<typeof sessionTokenSelector> =
+    yield* select(sessionTokenSelector);
+
+  const sessionToken: SagaCallReturnType<typeof authenticationSaga> =
+    previousSessionToken
+      ? previousSessionToken
+      : yield* call(authenticationSaga);
```

Comment:
If we call the `authenticationSaga` without blocking the current saga this would be stuck here and continue after the `authenticationSaga` execution would be over.

My suggestion is to handle this case properly and end the execution with a return in case we ask to run the authenticationSaga

Reply:
You are right! I've found that this piece of code taken from the startup saga, it doesn't fit well here. It's much better to call a session expired action to trigger the right authentication flow if for some reasons we come here without a session token.